src/K9OCRS/Controllers/AccountController.cs

```diff
@@ -254,11 +229,112 @@ public async Task<IActionResult> ChangePassword(PasswordReset passwordReset)
             }
-            //Create another controller that uses Request.QueryString to read token. If valid, load page, reset password.
             return Ok();
         }
 
+        [HttpPost("changeinfo")]
+        public async Task<IActionResult> ChangeInfo([FromBody] User newInfo)
+        {
+            User user = await connectionOwner.Use(conn =>
+            {
+                return dbOwner.Users.GetByID(conn, newInfo.ID);
+            });
+            user.Email = newInfo.Email;
+            user.FirstName = newInfo.FirstName;
+            user.LastName = newInfo.LastName;
+
+            var tasks = new List<Task>();
+            tasks.Add(connectionOwner.UseTransaction(async (conn, tr) =>
+            {
+                await dbOwner.Users.Update(conn, tr, user);
+                tr.Commit();
+            }));
+
+            await Task.WhenAll(tasks);
+            return Ok();
+        }
+
+        [HttpPost("changeinfoadmin")]
+        public async Task<IActionResult> ChangeInfoAdmin([FromBody] User newInfo)
+        {
+            User user = await connectionOwner.Use(conn =>
+            {
+                return dbOwner.Users.GetByID(conn, newInfo.ID);
+            });
+            user.Email = newInfo.Email;
+            user.FirstName = newInfo.FirstName;
+            user.LastName = newInfo.LastName;
+            user.UserRoleID = newInfo.UserRoleID;
+            user.ProfilePictureFilename = newInfo.ProfilePictureFilename;
+
+            var tasks = new List<Task>();
+            tasks.Add(connectionOwner.UseTransaction(async (conn, tr) =>
+            {
+                await dbOwner.Users.Update(conn, tr, user);
+                tr.Commit();
+            }));
+
+            await Task.WhenAll(tasks);
+            return Ok();
+        }
+
+        [HttpPost("createuser")] 
+        public async Task<IActionResult> CreateUser([FromBody] CreateUser accountInfo)
+        {
+            if(await ValidateEmailPassword(accountInfo.Email, accountInfo.Password))
+            {
+                var result = await connectionOwner.Use(conn =>
+                {
+                    User user = new();
+                    user.FirstName = accountInfo.First;
+                    user.LastName = accountInfo.Last;
+                    user.Email = accountInfo.Email;
+                    user.Password = GetHashedPassword(accountInfo.Password);
+                    user.UserRoleID = accountInfo.Role;
+                    return dbOwner.Users.Add(conn, user);
+                });
+                return Ok("Account added");
+            }
+            return StatusCode(400, "Failed to create user");
+        }
+
+        [HttpPost("getuser")]
+        public async Task<IActionResult> GetUser([FromBody] int id)
+        {
+            User user = await connectionOwner.Use(conn =>
+            {
+                return dbOwner.Users.GetByID(conn, id);
+            });
+            UserResult userResult = new UserResult(user);
```

Comment:
Make sure to include the storageBasePath from the service constants, it will still convert to a UserResult but it won't generate a url if it doesn't have the storageBasePath to use.

```suggestion
            UserResult userResult = new UserResult(user, serviceConstants.storageBasePath);
```